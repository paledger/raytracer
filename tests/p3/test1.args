pixeltrace ../resources/refract2.pov 640 480 295 265
