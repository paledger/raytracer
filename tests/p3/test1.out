Pixel: [295, 265] Color: (80, 5, 118)
o - Iteration type: Primary
|   Ray: {0 0 14} -> {-0.0509 0.05298 -0.9973}
|   Hit Object ID (1 - Sphere) at T = 11.14, Intersection = {-0.5673 0.5904 2.886}
|   Normal {-0.1891 0.1968 0.962}
|   Transformed Ray: {0 0 14} -> {-0.0509 0.05298 -0.9973}
|   Ambient: 0.1, 0, 0.1
|   Diffuse: 0.1157, 0, 0.1157
|   Specular: 0, 0, 0
|
|\
| o - Iteration type: Refraction
| |   Ray: {-0.5673 0.5904 2.885} -> {0.01087 -0.01131 -0.9999}
| |   Hit Object ID (1 - Sphere) at T = 5.796, Intersection = {-0.5043 0.5248 -2.91}
| |   Normal {-0.1681 0.1749 -0.9701}
| |   Transformed Ray: {-0.5673 0.5904 2.885} -> {0.01087 -0.01131 -0.9999}
| |   Ambient: 0.1, 0, 0.1
| |   Diffuse: 0, 0, 0
| |   Specular: 0, 0, 0
| |   Extra Info: into-object 
| |
| |\
| | o - Iteration type: Refraction
| | |   Ray: {-0.5042 0.5248 -2.911} -> {0.07256 -0.07552 -0.9945}
| | |   Hit Object ID (3 - Plane) at T = 59.91, Intersection = {3.843 -4 -62.5}
| | |   Normal {0 1 0}
| | |   Transformed Ray: {-0.5042 0.5248 -2.911} -> {0.07256 -0.07552 -0.9945}
| | |   Ambient: 0.08, 0.08, 0.32
| | |   Diffuse: 0.1139, 0.1139, 0.4557
| | |   Specular: 0, 0, 0
| | |   Extra Info: into-air 
| | |----
| | 
|  \
|   o - Iteration type: Reflection
|   |   Ray: {-0.5043 0.5248 -2.91} -> {0.3357 -0.3494 0.8748}
|   |   Hit Object ID (3 - Plane) at T = 12.95, Intersection = {3.843 -4 8.419}
|   |   Normal {0 1 0}
|   |   Transformed Ray: {-0.5043 0.5248 -2.91} -> {0.3357 -0.3494 0.8748}
|   |   Ambient: 0.08, 0.08, 0.32
|   |   Diffuse: 0.1441, 0.1441, 0.5766
|   |   Specular: 0, 0, 0
|   |----
|   
 \
  o - Iteration type: Reflection
  |   Ray: {-0.5673 0.5904 2.886} -> {-0.4062 0.4227 0.8101}
  |   No intersection.
  |----
  

--------------------------------------------------------------------------------

